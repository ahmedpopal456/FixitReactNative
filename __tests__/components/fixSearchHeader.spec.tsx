import React from 'react';
import 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import FixSearchHeader from '../../src/components/fixSearchHeader';

describe('Fix Search Header', () => {
  const goBack = jest.fn();
  const canGoBack = jest.fn();
  canGoBack.mockImplementation(() => true);
  const props = {
    showBackBtn: false,
    navigation: {
      goBack,
      canGoBack,
      HomeScreen: undefined,
      SearchResultsScreen: {
        tags: [],
      },
      FixRequestMetaStep: undefined,
      FixRequestDescriptionStep: undefined,
      FixRequestSectionsStep: undefined,
      FixRequestImagesLocationStep: undefined,
      FixRequestScheduleStep: undefined,
      FixRequestReview: {
        passedFix: {},
        isFixCraftsmanResponseNotification: false,
      },
      FixSuggestChanges: { passedFix: {} },
      FixSuggestChangesReview: { passedFix: {}, cost: '', comments: '' },
    },
    screenTitle: 'screen title',
  };

  it('renders correctly', async () => {
    const fixSearchHeader = renderer.create(
      <FixSearchHeader {...props} />,
    ).toJSON();
    expect(fixSearchHeader).toMatchSnapshot();
  });

  it('handle go back', async () => {
    const wrapper = shallow(
      <FixSearchHeader {...props} />,
    );
    wrapper.instance().handleGoBack();
    expect(goBack).toHaveBeenCalledTimes(1);
  });
});
