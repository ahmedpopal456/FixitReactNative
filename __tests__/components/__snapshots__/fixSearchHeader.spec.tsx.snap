// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fix Search Header renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EEEEEE",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "elevation": 2,
      "flex": 0,
      "flexDirection": "column",
      "padding": 10,
      "paddingTop": 60,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#333",
        "borderRadius": 8,
        "margin": 10,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      Search bar goes here
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 40,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        color="dark"
        fontSize={24}
        fontWeight={400}
        justification="left"
        padding={3}
        style={
          Array [
            Object {
              "color": "#1D1F2A",
              "flexDirection": "row",
              "fontFamily": "Raleway",
              "fontSize": 24,
              "fontWeight": "400",
              "paddingBottom": 3,
              "paddingLeft": 3,
              "paddingRight": 3,
              "paddingTop": 3,
              "textAlign": "left",
            },
          ]
        }
      >
        screen title
      </Text>
    </View>
  </View>
</View>
`;
